src/index.js

```diff
@@ -75,14 +70,29 @@ module.exports = {
   included(app) {
     this._super.included.apply(this, arguments);
 
-    // see: https://github.com/ember-cli/ember-cli/issues/3718
-    if (typeof app.import !== 'function' && app.app) {
-      // eslint-disable-next-line no-param-reassign
-      app = app.app;
+    if (app.parent) {
+      // we are being used by an addon (which is not to be confused
+      // with being used by an addon's dummy app!). In this case,
+      // app.options.svgJar is coming from the addon's index.js. The
+      // viewer defaults to off, and our default source for SVGs is
+      // the addon's public directory.
+      this.svgJarOptions = buildOptions(
+        app.options.svgJar,
+        false,
+        path.join(app.root, app.treePaths.public)
+      );
+    } else {
+      // we are being used by an app (which may also include the dummy
+      // app inside of an addon).  In this case, app.options.svgJar is
+      // coming from the app's ember-cli-build.js, the viewer defaults
+      // to "on in development", and our default source for icons is
+      // the app's public directory.
+      this.svgJarOptions = buildOptions(
+        app.options.svgJar,
+        app.env === 'development',
+        app.options.trees.public
```

Comment:
This takes care of automatically finding both `public` and `tests/dummy/public` depending on whether we are a normal app or an addon's dummy app.

Sometimes the thing we find here will already be a Broccoli tree (not just a string).